Fix timestamp parsing of github hook submissions.

Sometimes the timestamp has format "2013-10-31T22:53:56Z" (note "Z" at
the end, rather than the +HH:MM that Buildbot was expecting).

--- github.py.orig	2015-03-17 11:48:46.811526878 +0100
+++ github.py	2015-03-17 11:50:02.103081726 +0100
@@ -54,10 +54,15 @@
     
 def convertTime(myTestTimestamp):
     #"1970-01-01T00:00:00+00:00"
-    matcher = re.compile(r'(\d\d\d\d)-(\d\d)-(\d\d)T(\d\d):(\d\d):(\d\d)([-+])(\d\d):(\d\d)')
+    matcher = re.compile(r'(\d\d\d\d)-(\d\d)-(\d\d)T(\d\d):(\d\d):(\d\d)([-+Z])((\d\d):(\d\d))?')
     result  = matcher.match(myTestTimestamp)
-    (year, month, day, hour, minute, second, offsetsign, houroffset, minoffset) = \
+    (year, month, day, hour, minute, second, offsetsign, hourmin, houroffset, minoffset) = \
         result.groups()
+    if offsetsign == 'Z':
+        houroffset = "00"
+        minoffset = "00"
+        offsetsign = '+'
+
     if offsetsign == '+':
         offsetsign = 1
     else:
